packages/core/database/packup.config.ts

```diff
@@ -2,6 +2,21 @@
 import { defineConfig } from '@strapi/pack-up';
 
 export default defineConfig({
-  externals: ['crypto', 'node:async_hooks', 'node:path', 'path', 'stream', 'timers', 'tty'],
+  externals: [
+    'crypto',
+    'node:async_hooks',
+    'node:path',
+    'path',
+    'stream',
+    'timers',
+    'tty',
+    /**
+     * Knex dependencies, if we don't mark these as external
+     * they will be included in the bundle which means they
+     * will fail...
+     */
+    'knex/lib/query/querybuilder',
```

Comment:
Would it work if we only put knex ? 

Have you looked into https://www.npmjs.com/package/rollup-plugin-node-externals. I used it in the past.

Reply:
It'll already set `knex` as an external because it's in the dependencies, so i did find it weird that it wasn't resolving the full path, but equally i'd say knex is bundled slightly oddly.

RE: node externals, i've used it a couple of times, I think this could be a good enhancement to include overall, i'd like to look into whether to use that package or write something _relatively_ quick like a function for the external option.